dataclasses_json/api.py

```diff
@@ -25,11 +29,181 @@ class LetterCase(Enum):
+            expected_value = unknown
+
+        known[catch_all_field.name] = expected_value
+        return known
+
+    @staticmethod
+    def handle_to_dict(obj, kvs: Dict[Any, Any]) -> Dict[Any, Any]:
+        catch_all_field = CatchAllUndefinedParameters._get_catch_all_field(obj)
+        undefined_parameters = kvs.pop(catch_all_field.name)
+        if isinstance(undefined_parameters, dict):
+            kvs.update(undefined_parameters)  # If desired handle letter case here
+        return kvs
+
+    @staticmethod
+    def handle_dump(obj) -> Dict[Any, Any]:
+        catch_all_field = CatchAllUndefinedParameters._get_catch_all_field(cls=obj)
+        return getattr(obj, catch_all_field.name)
+
+    @staticmethod
+    def create_init(obj) -> Callable:
+        original_init = obj.__init__
+        init_signature = inspect.signature(original_init)
+
+        @functools.wraps(obj.__init__)
+        def _catch_all_init(self, *args, **kwargs):
+            known_kwargs, unknown_kwargs = CatchAllUndefinedParameters._separate_defined_undefined_kvs(obj, kwargs)
+            num_params_takeable = len(init_signature.parameters) - 1  # don't count self
+            if CatchAllUndefinedParameters._get_catch_all_field(obj).name not in known_kwargs:
+                num_params_takeable -= 1
+            num_args_takeable = num_params_takeable - len(known_kwargs)
+
+            args, unknown_args = args[:num_args_takeable], args[num_args_takeable:]
+            bound_parameters = init_signature.bind_partial(self, *args, **known_kwargs)
+            bound_parameters.apply_defaults()
+
+            unknown_args = {f"_UNKNOWN{i}": v for i, v in enumerate(unknown_args)}
+            arguments = bound_parameters.arguments
+            arguments.update(unknown_args)
+            arguments.update(unknown_kwargs)
+            arguments.pop("self", None)
+            final_parameters = CatchAllUndefinedParameters.handle_from_dict(obj, arguments)
+            original_init(self, **final_parameters)
+
+        return _catch_all_init
+
+    @staticmethod
+    def _get_catch_all_field(cls) -> Field:
+        catch_all_field = None
+        for field in fields(cls):
+            if field.type == Optional[CatchAllVar]:
+                if catch_all_field is not None:
+                    raise UndefinedParameterError(
+                        f"Multiple catch-all fields supplied: {catch_all_field.name, field.name}.")
+                catch_all_field = field
+        if catch_all_field is None:
+            raise UndefinedParameterError("No field of type dataclasses_json.CatchAll defined")
+        return catch_all_field
+
+
+class UndefinedParameters(Enum):
+    """
+    Choose the behavior what happens when an undefined parameter is encountered during class initialization.
+    """
+    INCLUDE = CatchAllUndefinedParameters
+    RAISE = RaiseUndefinedParameters
+    EXCLUDE = IgnoreUndefinedParameters
+    DEFAULT = None  # Same as not specifying a parameter
```

Comment:
this case is unnecessary, given we are modeling the type as `Optional` in the API

Reply:
Deleted it